libs/mocks/src/connectors_gateway_routers.rs

```diff
@@ -0,0 +1,33 @@
+use cfg_traits::connectors::Router;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::dispatch::DispatchResult;
+use scale_info::TypeInfo;
+use sp_std::{default::Default, marker::PhantomData};
+
+use crate::MessageMock;
+
+#[derive(Default, Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct DomainRouterMock<T: frame_system::Config> {
+	_marker: PhantomData<T>,
+}
+
+impl<T: frame_system::Config> DomainRouterMock<T> {
+	pub fn new() -> Self {
+		Self {
+			_marker: PhantomData::default(),
+		}
+	}
+}
+
+impl<T: frame_system::Config> Router for DomainRouterMock<T> {
+	type Message = MessageMock;
+	type Sender = T::AccountId;
+
+	fn init(&self) -> DispatchResult {
+		Ok(())
+	}
+
+	fn send(&self, _sender: Self::Sender, _message: Self::Message) -> DispatchResult {
+		Ok(())
+	}
+}
```

Comment:
I would've loved to have actual mocking capabilities here, but a router is not a pallet. Would it make sense to treat it like one for the purpose of tests and adapt this accordingly?

Reply:
From the `connectors-gateway` perspective, it doesn't know that traits that currently are pallets, really are. So yes, I think it's perfectly fine to add this pallet as a mock-builder independently of the actual implementation you will give it later.

Basically, the rule for using `mock-builder` would be: Any trait used in any associated type of the `Config` trait of a pallet (as `connector-gateway` does) can be mocked using `mock-builder`.

Reply:
OK, then I will go ahead and change this so we have a "real" mock, thank you!

Reply:
@lemunozm Done, I had to build it as a small wrapper on top of the pallet, please let me know what you think about that. Cheers!

Reply:
Thanks for the change! I've answered you inline below

Reply:
Ok, you were right. Sorry because I didn't pay attention that the init/send methods using `self` were instead of static trait methods 🙏🏻. The idea of `mock-builder` fit only if methods of the trait are static. So I think the previous change you did is the correct one. Sorry for making you dizzy with this!

The correct definition will be: *Any trait **with static methods** used in any associated type of the Config trait of a pallet (as connector-gateway does) can be mocked using mock-builder.*